shared/src/popup.js

```diff
@@ -193,81 +216,60 @@ async function setup() {
     });
   });
 
-  function toggleAdvancedDisplay(forceState) {
-    const icons = eAdvanced.querySelectorAll('svg');
+  async function toggleAdvancedDisplay(forceState) {
+    const icons = advancedToggle.querySelectorAll('svg');
     const showSettingsIcon = icons[0];
     const closeSettingsIcon = icons[1];
 
-    if (forceState === 'close' || eTokenDiv.style.display === '') {
+    if (forceState === 'close' || tokenDiv.style.display === '') {
       showSettingsIcon.style.display = '';
       closeSettingsIcon.style.display = 'none';
-      eTokenDiv.style.display = 'none';
-      if (eTokenInput.value) {
-        eSummarize.style.display = '';
+      tokenDiv.style.display = 'none';
+      if (tokenInput.value) {
+        const hasPermissions = await browser.permissions.contains({
+          permissions: ['activeTab'],
+        });
```

Comment:
One interesting, hard to debug fact is that Firefox will complain if you try to check for permissions and request them on the same event loop. That's why I ended up having to create a separate button/section (initially I just had the selects hide and the summarize button change its text). Chrome didn't care, though.